### PR TITLE
Add example input/outputs in Rapid7 InsightVM plugin

### PR DESCRIPTION
**Pull request by r7-kszczepanskagorna**
## Proposed Changes
Add example input/outputs in Rapid7 InsightVM plugin

### Description

Add example input/outputs which reflected real examples and match to the example outputs.
Ticket - MC-393

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../../tools/Makefiles/Helpers.mk ../../tools/Makefiles/Colors.mk
--
[*] Running validators
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Executing validator CloudReadyValidator
[*] Executing validator UnapprovedKeywordsValidator
318
[*] Plugin successfully validated!

----
[*] Total time elapsed: 11749.674ms
```

<summary>
make validate
</summary>
</details>

<details>

```
[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Executing validator CloudReadyValidator
[*] Executing validator UnapprovedKeywordsValidator
318
[*] Executing validator ExceptionValidator
WARNING: Use of 'PluginException' or 'ConnectionTestException' is recommended when raising an exception.
violation: komand_rapid7_insightvm/util/resource_helpers.py: line,  117
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
WARNING: URLs found that return a 4xx code. Verify they are publicly accessible and if not, update with a working URL.
violation: plugin.spec.yaml[1747]: https://insightvm.example.com:3780
violation: plugin.spec.yaml[1750]: https://insightvm.example.com:3780
violation: help.md[255]: https://example.com:3780/api/3/vulnerabilities/ssh-default-account-admin-password-admin
violation: help.md[121]: https://ivm-console.rapid7.com:3780/api/3/solutions/adobe-flash-windows-upgrade-latest/prerequisites
violation: help.md[105]: https://ivm-console.rapid7.com:3780/api/3/assets/234/vulnerabilities/flash_player-cve-2017-11305/solution
violation: help.md[234]: https://example.com:3780/api/3/vulnerability_exceptions/1
violation: help.md[244]: https://example.com:3780/api/3/users/1
violation: help.md[268]: https://example.com:3780/api/3/users/1
violation: help.md[117]: https://ivm-consolelax.rapid7.com:3780/api/3/solutions/adobe-flash-windows-upgrade-latest
violation: help.md[1309]: https://ivm-console-test.vuln.lax.rapid7.com:3780/...
violation: help.md[1313]: https://ivm-console-test.vuln.lax.rapid7.com:3780/...
violation: help.md[125]: https://ivm-console.rapid7.com:3780/api/3/solutions/adobe-flash-windows-upgrade-latest/supersedes
violation: help.md[141]: https://ivm-console.rapid7.com:3780/api/3/solutions/adobe-flash-upgrade-28-0-0-126-windows
[*] Plugin successfully validated!

----
[*] Total time elapsed: 162774.99ms
```

<summary>
icon-validate --all . 
</summary>
</details>

### CODE REVIEW

File: plugins/rapid7_insightvm/help.md [hunk 1]
```diff
@@ -38,6 +38,45 @@ Example input:
 
 ### Actions
 
+#### Add User Asset Group Access
+
+This action is used to grant an user account access to an asset group by ID.
```
Comment (mrinehart-r7) [inline plugins/rapid7_insightvm/help.md:43]:
```suggestion
This action is used to grant a user account access to an asset group by ID.
```
This should ideally be corrected in the plugin.spec.yaml so the regeneration sticks - but, I can't find this action in the plugin.spec.yaml 🤔

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

**Review by emartin-merrill-r7 — COMMENTED** [review]
@jmcadams-r7 @mrinehart-r7 Here's the review we discussed 6.28

File: plugins/rapid7_insightvm/plugin.spec.yaml [hunk 2]
```diff
@@ -2028,7 +2028,7 @@ actions:
         title: Search Criteria
         description: Tag search Criteria - options documentation https://help.rapid7.com/insightvm/en-us/api/#section/Responses/SearchCriteria
```
[the remark this answers is not in the document]
Comment (r7-mgorny) [reply plugins/rapid7_insightvm/plugin.spec.yaml:2029]: @emartin-merrill-r7 In documentation there is:
https://help.rapid7.com/insightvm/en-us/api/#operation/findAssets
It show what user should use in fields:
"Filters used to match assets. See [Search Criteria](https://help.rapid7.com/insightvm/en-us/api/#section/Responses/SearchCriteria) for more information on the structure and format."

The "Search Criteria" is a link which link to: 
"https://help.rapid7.com/insightvm/en-us/api/#section/Responses/SearchCriteria".

Which link should we use?

**Comment by blaxminarayan-r7** [comment]
Closing this because the help.md was updated in a recent release

**Pull request closed without merging**